sql/src/test/java/io/crate/integrationtests/PostgresITest.java

```diff
@@ -601,6 +602,7 @@ public void testStatementThatResultsInParseError() throws Exception {
             stmt.executeQuery();
         }
     }
+    */
```

Comment:
I'd like to discuss this one as the error seems a bit odd in this case.

This test is now failing with Error `Cannot find data type: fro` instead of `mismatched input 'sys'`

I've tried a similar query in pg:
```
mydb=> select name fro test1;
2019-04-15 12:51:11.551 EEST [65047] ERROR:  syntax error at or near "test1" at character 17
2019-04-15 12:51:11.551 EEST [65047] STATEMENT:  select name fro test1;
ERROR:  syntax error at or near "test1"
LINE 1: select name fro test1;
```
It's not as clear I would have expected but it'd better than the type error.
So it seems that the error is handled differently.

What's your opinion on that?

Reply:
hmm that seems to behave really wrong, it should not try to resolve a data type. It seems that this parser rule is not even matched in this case
```
selectItem:
   expr (AS? ident)?  
```

Reply:
hmm. I think I should really make this rule strict compared to the other notations (`cast()`, `::`)

Reply:
@kovrus `selectItem` rule is matched, the issue is that it matches the `alternateCast` rule later on.

Reply:
yep, right

Reply:
@kovrus I finally restricted the rule to accept a stringLiteral which is actually the use case anyway